### PR TITLE
Checks for ffmpeg and git in PATH

### PR DESCRIPTION
**Pull request by Twentysix26**
Needs to be tested on various Windows systems and maybe needs a more helpful message in the git check. I'll just leave it here in the meantime.

### CODE REVIEW

**Comment by Kowlin** [comment]
Right now I cannot get this to work, Git is installed while Ffmpeg isn't in the directory.

As far as I can see without digging through the code I don't think its even checking for ffmpeg.

File: cogs/audio.py [hunk 1]
```diff
@@ -1876,6 +1876,19 @@ def voice_connected(self, server):
                 vc.audio_player.resume()
 
 
+def check_ffmpeg():
+    """Windows only"""
+    files = ("ffmpeg.exe", "ffplay.exe", "ffprobe.exe")
+    if os.name == "nt":
+        if "ffmpeg" in os.environ['PATH']:
```
Comment (Kowlin) [inline cogs/audio.py:1883]: I rewrote from line `1883` to the following,

``` python
        try:
            call(['git.exe'], stdout=DEVNULL, stderr=DEVNULL)
            return True
        except:
            return False
```

The only thing that needs to be done besides this is importing `DEVNULL` in `Subprocess`

**Comment by Fr6jDJF** [comment]
with C:\ffmpeg\bin ; in PATH Windows failed on me
:white_check_mark: C:>ffmpeg -version
:x: C:>ffplay -version
:x: C:>ffprobe -version

:white_check_mark: C:\ffmpeg\bin\ffplay -version
:white_check_mark: C:\ffmpeg\bin\ffprobe -version

A reboot solved this, but the code gave me true while it was not available for python
 :white_check_mark: https://github.com/Twentysix26/Red-DiscordBot/pull/341 returned True :warning: 
 :white_check_mark: subprocess.call(' ffmpeg -i "{0}" -acodec flac "{1}" ', shell=True)
 :x:subprocess.Popen("ffprobe", some_file, stdout = subprocess.PIPE, stderr = subprocess.STDOUT) ([WinError 2] not found)

When adding ffprobe in some data dir
:white_check_mark: subprocess.Popen("data/freesound/ffprobe", some_file, stdout = subprocess.PIPE, stderr = subprocess.STDOUT) 

The code below confirmed my not available status

``` python
def shutil_ffmpeg():
    """Windows only"""
    files = ("ffmpeg", "ffprobe", "ffplay", "git")
    if os.name == "nt":
        # Return the path to an executable which would be run if the given cmd was called. If no cmd would be called, return None.
        # When no path is specified, the results of os.environ() are used, returning either the “PATH” value or a fallback of os.defpath.
        if not all([shutil.which(f) for f in files]): 
            return False
        else:
            return True
    else:
        return True
```

**Comment by Kowlin** [comment]
Oh look who is being fancy with the check marks :P

**Comment by ghost** [comment]
Is there reason to not do this check on linux as well? if checking on linux is fine, it should simplify down to

``` python
def on_path(*executables):
    return all(shutil.which(f) for f in executables)
```

``` python
    if not on_path('ffmpeg','ffprobe','ffplay'):
        raise RuntimeError("You need ffmpeg to play audio.\n{}"
                           "".format(support_page))
```

``` python
    if not on_path('git'):
        raise RuntimeError("You don't have git in your PATH environment variable\n"
                           "That's necessary for the downloader to run\n")
```

`if not on_path` also reads really cleanly imo

**Review by ghost — APPROVED** [review]
Just trying something?

**Comment by Twentysix26** [comment]
This was mostly a proof of concept, will come up with something better.

**Pull request closed without merging**